latex_access/tests/test_preprocessor.py

```diff
@@ -0,0 +1,68 @@
+import os
+import unittest
+import tempfile
+import pickle
+
+try:
+    from latex_access import preprocessor
+except ImportError:
+    incompatible = True
+else:
+    incompatible = False
+
+HERE = os.path.abspath(os.path.dirname(__file__))
+
+
+@unittest.skipIf(incompatible, "This module is not compatible with version 3 of Python.")
+class TestPreprocessor(unittest.TestCase):
+    def setUp(self):
+        """Creates a preprocessor instance."""
+        self.preprocessor = preprocessor.preprocessor()
+
+    def test_table(self):
+        """Tests creation of table attribute."""
+        self.assertEqual(self.preprocessor.table, {})
+
+    def test_add(self):
+        """Test adding command to table."""
+        self.preprocessor.add('foo', 'bar')
+        self.assertEqual(self.preprocessor.table['foo'], ('bar'))
+
+    def test_add_from_string(self):
+        """Tests adding command with args."""
+        self.preprocessor.add_from_string('foo', 'args', '1#2#3')
+        self.assertEqual(self.preprocessor.table['foo'], ['args', '1', 2, '', 3, ''])
+
+    def test_write(self):
+        """Tests that preprocessor entries can be saved to a file."""
+        with tempfile.NamedTemporaryFile(delete=False) as file:
```

Comment:
Would it be possible to get rid of `delete=False`, so that this temporary file is automatically removed when the test ends, and move the block which unpickles the dictionary inside `with` block?

Reply:
It doesn't work when I try to modify it as you suggest and I'm not sure if it's worth investigating further.

Reply:
After some reading it turns out this won't work, since `preprocessor.write` calls `close` on the temporary file which deletes it from the file system. Sorry for the noise!